services/cache.js

```diff
@@ -27,23 +32,12 @@ const isInCache = (requestedUrl) => {
   return null
 }
 
-const handleCachedData = async (req, res, next) => {
-  if (req.method === 'GET' && config.isCacheEnabled) {
-    const url = req.url
-    const requestedUrl = await cacheDataClient.get(url)
-
-    if (isInCache(requestedUrl)) {
-      return res.status(200).send(JSON.parse(requestedUrl).data)
-    } else {
-      const returnedResponse = res.send
-      res.send = (body) => {
-        saveToCache(url, body)
-        res.send = returnedResponse
-        res.send(body)
-      }
+const clearOutdatedCache = (url, method) => {
+  config.cache.map((cachedReq) => {
+    if (pathToRegexp(cachedReq.requestUrl).exec(url) && cachedReq.method === method) {
+      cachedReq.cacheToClear.map((cachedUrl) => cacheDataClient.del(cachedUrl))
```

Comment:
please make sure the request url saved in the cache from the middleware (req.url) is exactly the same as the url you put in config. i suspect one is relative and the other is full path

Reply:
Switched to req.path (as it gives relative path similar to the one in config.)